CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.8.1)
+project(quick LANGUAGES NONE)
+
+
+if(NOT INSIDE_AMBER)
+
+	# initialization and include paths
+	#-------------------------------------------------------------
+	include(cmake/AmberBuildSystemInit.cmake NO_POLICY_SCOPE)
+
+	#now enable the languages
+	enable_language(C CXX Fortran)
+
+	include(AmberBuildSystem2ndInit)
+	
+	include(MPIConfig)
+	include(CudaConfig)
+
+	set(LINALG_LIBS_REQUIRED TRUE)
+	set(NEEDED_3RDPARTY_TOOLS blas lapack mkl)
+	set(REQUIRED_3RDPARTY_TOOLS )
+	set(BUNDLED_3RDPARTY_TOOLS blas lapack)
+	include(3rdPartyTools)
+	
+	include(CompilerFlags)
+
+	
+	#CPack setup
+	# --------------------------------------------------------------------
+	set(PACKAGE_NAME "quick")
+	set(PACKAGE_FILENAME "q")
+
+	set(BUNDLE_IDENTIFIER org.merzlab.quick)
+	set(BUNDLE_SIGNATURE QUIK)
+	include(Packaging)
+
+	# Fortran module dirs
+	# --------------------------------------------------------------------
+	set(XC_SERIAL_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/serial)
+	set(XC_MPI_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/mpi)
+	set(XC_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/gpu)
+
+	set(QUICK_SERIAL_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/serial)
+	set(QUICK_MPI_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/mpi)
+	set(QUICK_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/gpu)
+	set(QUICK_MPI_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/mpi_gpu)
+
+endif()
+
+option(NOF "Disables the compilation of QUICK's time consuming f functions in the ERI code of cuda version. Not recommended for production." FALSE)
+
+# C/Fortran compiler flags
+# --------------------------------------------------------------------
+
+# add -DDEBUG in debug configuration
+set_property(DIRECTORY . PROPERTY COMPILE_DEFINITIONS $<$<CONFIG:Debug>:DEBUG>)
+
+
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "Intel")
+	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -traceback")
+endif()
+
+
+if("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU")
+	set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -cpp")
+	add_definitions(-DGNU)
```

Comment:
If `target_compile_definitions` was used here, `-DGNU` would be `GNU`. But if `target_compile_options` was used then nothing changes. Is that correct?

Reply:
yep!